custom_components/frigate/media_source.py

```diff
@@ -1360,20 +1334,33 @@ def _browse_recording_folders(
                     thumbnail=None,
                 )
             )
+
         return base
 
-    def _browse_recordings(
-        self, identifier: RecordingIdentifier, recordings: list[dict[str, Any]]
+    def _get_recording_hours(
+        self, identifier: RecordingIdentifier, recording_days: list[dict[str, Any]]
     ) -> BrowseMediaSource:
         """Browse Frigate recordings."""
         base = self._get_recording_base_media_source(identifier)
+        hour_items = [
+            hours["hours"]
+            for hours in recording_days
+            if hours["day"] == identifier.year_month_day
+        ]
+
+        for hour_data in hour_items[0]:
```

Comment:
Why are we taking only the first item here? If this is a flat list with only a single item in it, best to avoid putting it in a list above.

Reply:
It is a flat list, I just wasn't sure of an easier way to pick the day that matched the identifier. Let me know if you have any ideas.

Reply:
Maybe this is slightly cleaner, although the for statement isn't the most readable:

```
chosen_day = filter(lambda day: day['day'] == identifier.year_month_day, recording_days)

for hour_day in (chosen_day and chosen_day[0]) or []:
  [...]
```

(This also correctly handles the case where an identifier is provided that has no matching data)

Reply:
I think my solution should be clean, let me know what you think